server/src/main/java/io/crate/planner/optimizer/joinorder/JoinGraph.java

```diff
@@ -136,7 +137,7 @@ public Set<Edge> edges(LogicalPlan node) {
     }
 
     public static JoinGraph create(LogicalPlan plan, UnaryOperator<LogicalPlan> resolvePlan) {
-        return plan.accept(new GraphBuilder(resolvePlan), new HashMap<>());
+        return plan.accept(new GraphBuilder(resolvePlan), new LinkedHashMap<>());
```

Comment:
Why is this now order sensitive?

Reply:
This is not only a testing issue. This guarantees that the new query plan always has the same order of symbols and join conditions.